Feature advanced search

alihamuh opened the pull request.

alihamuh committed:
web: ui and some functuanality is added.

alihamuh committed:
web: some minor bug-fixes

alihamuh committed:
web: started buidling search engine

alihamuh committed:
web: filter search completed

alihamuh committed:
web: added core functions for search

alihamuh committed:
web: refactored code

thecodrr commented on apps/web/src/navigation/routes.js:
```diff
@@ -212,13 +212,13 @@ const routes = {
+    component: <Search />,
     buttons: {
       back: {
-        title: `Go back to ${type}`,
+        title: `Go back to test`,
         action: () => window.history.back()
       }
```
This button should be removed.

thecodrr commented on package.json:
```diff
@@ -52,7 +52,10 @@
   "dependencies": {
     "@nrwl/nx-cloud": "^14.6.2",
-    "eslint-plugin-header": "^3.1.1"
+    "eslint-plugin-header": "^3.1.1",
+    "liqe": "^3.6.0",
+    "react-datepicker": "^4.10.0",
+    "react-fast-compare": "^3.2.0"
```
What are these libraries doing here? This is the global `package.json`.

thecodrr commented on apps/web/src/components/search/suggestionrow.js:
```diff
@@ -0,0 +1,57 @@
+        ml={1}
+        sx={{ textOverflow: "ellipsis" }}
+      >
+        {props.item.main2}
```
What's `main2`?

thecodrr commented on apps/web/src/components/search/suggestionrow.js:
```diff
@@ -0,0 +1,57 @@
+        ml={1}
+        sx={{ textOverflow: "ellipsis" }}
+      >
+        {props.item.main1}
```
What's `main1`?

thecodrr commented on apps/web/src/views/search.js:
```diff
@@ -17,137 +17,74 @@ You should have received a copy of the GNU General Public License
-      setResults(results);
-    })();
-  }, [nonce, type, context]);
+  useEffect(() => {}, []);
```
What's this for?

thecodrr commented on apps/web/src/views/search.js:
```diff
@@ -17,137 +17,74 @@ You should have received a copy of the GNU General Public License
+  useEffect(() => {}, []);
 
-  if (!title) return hardNavigate("/");
+  const type = "notes";
```
Why fix it to `notes`?

thecodrr commented on apps/web/src/views/search.js:
```diff
@@ -17,137 +17,74 @@ You should have received a copy of the GNU General Public License
     <>
       <Text variant="subtitle" mx={2}>
-        Searching {title}
+        Searching {type}
```
Just say `Searching...`

thecodrr commented on apps/web/src/views/search.js:
```diff
@@ -160,22 +97,75 @@ function Search({ type }) {
-              }
-            />
+        <>
+          <Flex mx={2}>
+            {results.map((result, index) =>
+              result.length > 0 ? (
+                <Button
+                  variant="tertiary"
+                  sx={{
+                    borderColor:
+                      index === selectedResult ? "primary" : undefined
+                  }}
+                  mr={1}
+                  onClick={() => {
+                    setSearchItem(result);
+                    setSelectedResult(index);
+                  }}
+                >
+                  {`${result[0].type}s`}
+                </Button>
+              ) : (
+                []
+              )
+            )}
+          </Flex>
```
What's this for?

thecodrr commented on apps/web/src/views/search.js:
```diff
@@ -160,22 +97,75 @@ function Search({ type }) {
+              )
+            )}
+          </Flex>
+          {searchItem.length > 0 ? (
+            <SearchResults
+              title={capitalizeFirstLetter(`${searchItem[0].type}s`)}
+              context={context}
+              type={`${searchItem[0].type}s`}
+              results={searchItem}
+            ></SearchResults>
+          ) : (
+            <Text variant="subtitle" mx={2}>
+              {"No items to show"}
+            </Text>
           )}
```
This can already be done via ListContainer. No need to reinvent the wheel.

thecodrr committed:
core: fix hang when resolving recursive nested notebooks

thecodrr committed:
common: bulk resolve notebooks & tags

ammarahm-ed committed:
mobile: remove READ_MEDIA_IMAGES permission

thecodrr committed:
web: move importer into web/desktop app (#7472)

Signed-off-by: 01zulfi <85733202+01zulfi@users.noreply.github.com>

thecodrr committed:
web: refresh app after import

01zulfi committed:
web: make nav menu header draggable (#7985)

* web: make part of nav menu header draggable
Signed-off-by: 01zulfi <85733202+01zulfi@users.noreply.github.com>

* Update apps/web/src/app.css

Signed-off-by: Abdullah Atta <thecodrr@protonmail.com>

* Update apps/web/src/components/navigation-menu/index.tsx

Signed-off-by: Abdullah Atta <thecodrr@protonmail.com>

---------

Signed-off-by: Abdullah Atta <thecodrr@protonmail.com>
Co-authored-by: Abdullah Atta <thecodrr@protonmail.com>

ammarahm-ed committed:
mobile: fix default font setting not followed

thecodrr committed:
web: open settings with importer page instead of importer.notesnook.com

thecodrr committed:
web: update @notesnook-importer/core to 2.2.0

01zulfi committed:
web: add support for archiving notes (#7810)

* web: archive notes
Signed-off-by: 01zulfi <85733202+01zulfi@users.noreply.github.com>

* global: arhives -> archived
Signed-off-by: 01zulfi <85733202+01zulfi@users.noreply.github.com>

thecodrr committed:
web: update @notesnook-importer/core to 2.2.1

ammarahm-ed committed:
mobile: fix typing in editor title input

01zulfi committed:
web: fix notestore is not defined error (#8012)

Signed-off-by: 01zulfi <85733202+01zulfi@users.noreply.github.com>

thecodrr committed:
web: update @notesnook-imoporter/core to 2.2.2

thecodrr committed:
web: bump version to 3.2.0-beta.0

ammarahm-ed committed:
mobile: fix cursor bug on ios

ammarahm-ed committed:
mobile: fix crash in list

ammarahm-ed committed:
mobile: fix crash on delete item

ammarahm-ed committed:
mobile: release v3.1.1

ammarahm-ed committed:
mobile: fix crash when going from trash -> search

ammarahm-ed committed:
ci: fix build ios

Signed-off-by: Ammar Ahmed <40239442+ammarahm-ed@users.noreply.github.com>

ammarahm-ed committed:
ci: fix build ios

Signed-off-by: Ammar Ahmed <40239442+ammarahm-ed@users.noreply.github.com>

ammarahm-ed committed:
ci: fix build ios

Signed-off-by: Ammar Ahmed <40239442+ammarahm-ed@users.noreply.github.com>

01zulfi committed:
core: segregate logic for note headline and note headline title generation (#7968)

* core: segregate logic for note headline and note headline title generation
Signed-off-by: 01zulfi <85733202+01zulfi@users.noreply.github.com>

* core: minor refactor note headline title
Signed-off-by: 01zulfi <85733202+01zulfi@users.noreply.github.com>

01zulfi committed:
editor: improve hover link popup position (#8011)

* move it down so it doesn't block the link

Signed-off-by: 01zulfi <85733202+01zulfi@users.noreply.github.com>

01zulfi committed:
web: fix list pane not expanding (#7963)

* web: fix list pane not expanding
* fix list pane not expanding to saved width on reload
* fix list pane not expanding to saved width on collapse

Signed-off-by: 01zulfi <85733202+01zulfi@users.noreply.github.com>

* web: collapse pane and set size to 0 when applicable
Signed-off-by: 01zulfi <85733202+01zulfi@users.noreply.github.com>

---------

Signed-off-by: 01zulfi <85733202+01zulfi@users.noreply.github.com>

01zulfi committed:
web: fix focus last active tab method (#8034)

Signed-off-by: 01zulfi <85733202+01zulfi@users.noreply.github.com>

01zulfi committed:
web: fix automatic backup reminder setting reverting to never (#8037)

Signed-off-by: 01zulfi <85733202+01zulfi@users.noreply.github.com>

01zulfi committed:
desktop: upgrade electron to v34.0.0 (#8038)

Signed-off-by: 01zulfi <85733202+01zulfi@users.noreply.github.com>

thecodrr committed:
desktop: update packages

01zulfi committed:
web: assign accent color to links (#8041)

Signed-off-by: 01zulfi <85733202+01zulfi@users.noreply.github.com>

thecodrr committed:
desktop: only show native menus on macos

thecodrr committed:
desktop: remove icon from native menu

thecodrr committed:
desktop: no need to externalize @resvg/resvg-js

thecodrr committed:
desktop: show native menus where possible

01zulfi committed:
web: add more archive test cases (#8040)

Signed-off-by: 01zulfi <85733202+01zulfi@users.noreply.github.com>

thecodrr committed:
core: wrap query with invalid symbols in quotes

thecodrr committed:
core: fix db migration

thecodrr committed:
core: update sqlite-better-trigram to v0.0.3

01zulfi committed:
core: fix trash cleanup interval setting retrieval (#8045)

Signed-off-by: 01zulfi <85733202+01zulfi@users.noreply.github.com>

ammarahm-ed committed:
mobile: update better trigram

thecodrr committed:
core: fix warnings in tests

thecodrr committed:
core: strip html when searching notes

this commit also properly fixes search queries containing
words less than 3 characters in length

ammarahm-ed committed:
mobile: add sqlite3-fts5-html module

thecodrr committed:
core: remove @orama/stopwords

thecodrr committed:
core: fix short queries search

thecodrr committed:
web: update wa-sqlite to use sqlite3-fts5-html for stripping html

thecodrr committed:
desktop: install sqlite3-fts5-html

thecodrr committed:
core: fix query-transformer tests

thecodrr committed:
web: bump version to 3.2.0-beta.1

ammarahm-ed committed:
mobile: archive notes

thecodrr committed:
global: update package-lock.json

thecodrr committed:
global: fix dependencies version differences

thecodrr committed:
web: update package-lock.json

thecodrr committed:
desktop: regenerate package-lock.json

thecodrr committed:
desktop: fix flatpak sources generation

sbrannstrom committed:
docs: add missing '-' in build-steps for MacOS and Linux (AppImage)

thecodrr committed:
desktop: downgrade playwright

thecodrr committed:
misc: include @notesnook/theme package when generating flatpak sources

ammarahm-ed committed:
mobile: add option to disable auto update check

ammarahm-ed committed:
mobile: fix error in build

thecodrr committed:
global: regenerate package-lock.json for theme & ui packages

thecodrr committed:
desktop: update patched better-sqlite3-multiple-ciphers to 11.6.1

ammarahm-ed committed:
editor: fix alignment items in callout heading (#8101)

* editor: fix alignment items in callout heading

* Update packages/editor/styles/styles.css

Signed-off-by: Abdullah Atta <thecodrr@protonmail.com>

---------

Signed-off-by: Abdullah Atta <thecodrr@protonmail.com>
Co-authored-by: Abdullah Atta <abdullahatta@streetwriters.co>

01zulfi committed:
web: press shift+enter to create note in quick open (#8068)

Signed-off-by: 01zulfi <85733202+01zulfi@users.noreply.github.com>

01zulfi committed:
web: add support for removing recents in command palette using delete key (#8069)

Signed-off-by: 01zulfi <85733202+01zulfi@users.noreply.github.com>

01zulfi committed:
core: fix db.settings.getTrashCleanupInterval return type (#8076)

Signed-off-by: 01zulfi <85733202+01zulfi@users.noreply.github.com>

thecodrr committed:
core: fix `expected previously executed migration 2 to be at index 1 but
10 was found in its place.`

thecodrr committed:
web: bump version to 3.2.0-beta.2

01zulfi committed:
web: add support for drag/dropping notes on tab strip (#8090)

Signed-off-by: 01zulfi <85733202+01zulfi@users.noreply.github.com>

01zulfi committed:
web: fix readonly note not locking when vault is locked (#8092)

Signed-off-by: 01zulfi <85733202+01zulfi@users.noreply.github.com>

01zulfi committed:
web: persist double spaced lines config on logout (#8097)

Signed-off-by: 01zulfi <85733202+01zulfi@users.noreply.github.com>

ammarahm-ed committed:
mobile: fix crash in settings -> user subscription

thecodrr committed:
core: add support for sortable search

thecodrr committed:
web: add support for sorting in search

thecodrr committed:
web: show date created on search results if sort by is set to date created

ammarahm-ed committed:
mobile: add line break when appending to note

ammarahm-ed committed:
mobile: fix new reminder title/headline is not autofill

thecodrr committed:
web: update sqlite3-fts5-html to v0.0.3

thecodrr committed:
core: add a new migration to reindex fts5

ammarahm-ed committed:
mobile: support sorting search results

ammarahm-ed committed:
mobile: update fts5-html lib

ammarahm-ed committed:
mobile: fix archive count not shown

ammarahm-ed committed:
mobile: handle crash in background on device boot

thecodrr committed:
core: add support for highlighting search results

thecodrr committed:
core: bring back search result sorting for notes

thecodrr committed:
core: fix title matches not merged when query contains short tokens

thecodrr committed:
core: improve search results highlighting

thecodrr committed:
core: add console.time tracing for notes search

thecodrr committed:
web: add support for highlighting in search results

ammarahm-ed committed:
mobile: upgrade to RN 0.77.2

ammarahm-ed committed:
mobile: update patches

01zulfi committed:
web: add 'move to top' for sub-notebooks (#8066)

Signed-off-by: 01zulfi <85733202+01zulfi@users.noreply.github.com>

thecodrr committed:
core: add raw html content with highlighting in results

thecodrr committed:
editor: add support for nn-search-result node

thecodrr committed:
web: add full support for highlighting results in notes search

thecodrr committed:
core: do not allow empty titles

thecodrr committed:
web: fix flicker when programatically updating title

thecodrr committed:
core: improve prefix/suffix generation for matches

thecodrr committed:
web: improve search results ui

thecodrr committed:
web: use same search ux for notes everywhere

thecodrr committed:
web: fix note not opening from search results if its already opened

thecodrr committed:
web: fix note not scrolling to selected match

01zulfi committed:
editor: support ctrl/cmd+click to open internal links in new tab (#8131)

Signed-off-by: 01zulfi <85733202+01zulfi@users.noreply.github.com>

thecodrr committed:
editor: make search results style toggleable

thecodrr committed:
web: hide search results on editor focus & search cancel

thecodrr committed:
web: localize search results count

ammarahm-ed committed:
mobile: search and sorting

thecodrr committed:
web: bump version to 3.2.0-beta.3

ammarahm-ed committed:
mobile: release v3.2.0-beta.0

thecodrr committed:
core: bring back old notes search

thecodrr committed:
web: use `notesWithHighlighting` for full notes search

ammarahm-ed committed:
mobile: fix note search

ammarahm-ed committed:
mobile: release v3.2.0-beta.1

thecodrr committed:
core: bring back `limit` in lookup

thecodrr committed:
web: fix various type errors causing crashes

thecodrr committed:
web: fix editor not jumping to search result

thecodrr committed:
ci: build desktop app on ubuntu-22-04

this fixes `Error: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.38'
not found` error

thecodrr committed:
web: bump version to v3.2.0-beta.4

thecodrr committed:
core: filter out stop words in search query

ammarahm-ed committed:
core: fix stop-words import

ammarahm-ed committed:
mobile: fix tab gets focused in loop

thecodrr committed:
desktop: enable regexp in sqlite

thecodrr committed:
web: enable regexp in sqlite

thecodrr committed:
web: use plain list instead of tree for showing search results

thecodrr committed:
core: install @notesnook/intl

thecodrr committed:
core: use intl for showing search results count

thecodrr committed:
core: speed up search

thecodrr committed:
web: bump version to 3.2.0-beta.5

thecodrr committed:
desktop: do not gitignore *.patch

thecodrr committed:
desktop: add `SQLITE_ENABLE_REGEXP` define to better-sqlite3

ammarahm-ed committed:
mobile: enable regexp in sqlite

thecodrr committed:
web: fix click on match no opening note at desired location

thecodrr committed:
core: fix phrase queries not getting highlighted

thecodrr committed:
core: auto expand outline & callout in search result

thecodrr committed:
web: discard results if query changed during search

ammarahm-ed committed:
mobile: skip rendering if query changed during search

ammarahm-ed committed:
mobile: release new beta

01zulfi committed:
editor: remove styling for sibling elements of double spaced paragraphs (#8144)

Signed-off-by: 01zulfi <85733202+01zulfi@users.noreply.github.com>

ammarahm-ed committed:
mobile: remove patch

01zulfi committed:
web: add support for setting a tag as default (#8060)

Signed-off-by: 01zulfi <85733202+01zulfi@users.noreply.github.com>

thecodrr committed:
web: bump version to 3.2.0-beta.6

thecodrr committed:
ci: fix `prerelease` not setting correctly on gh release

ammarahm-ed committed:
mobile: update fts5-html patch

ammarahm-ed committed:
mobile: fix ios build on rn 77

ammarahm-ed committed:
mobile: fix android build

thecodrr committed:
core: only expand matching callout & outline list items

thecodrr committed:
core: sort content matches

ammarahm-ed committed:
mobile: update lockfiles

ammarahm-ed committed:
core: skip decodine entities when parsing html

thecodrr committed:
core: update htmlparser2

thecodrr committed:
core: fix lookup tests

ammarahm-ed committed:
mobile: fix key missing warning in search-result item

ammarahm-ed committed:
mobile: update deps

thecodrr committed:
core: remove porter tokenizer for stemming

it didn't work very well. It was stemming things like "one" into "on"

thecodrr committed:
core: fix false positives appearing in search

ammarahm-ed committed:
mobile: beta release

ammarahm-ed committed:
mobile: fix mmkv patch

thecodrr committed:
web: bump version to v3.2.0-beta.7

ammarahm-ed committed:
mobile: fix ci build

01zulfi committed:
web: add move notebook dialog (#8099)

Signed-off-by: 01zulfi <85733202+01zulfi@users.noreply.github.com>
Co-authored-by: Abdullah Atta <abdullahatta@streetwriters.co>

01zulfi committed:
web: fix navigation commands in command palette (#8070)

Signed-off-by: 01zulfi <85733202+01zulfi@users.noreply.github.com>

ammarahm-ed committed:
mobile: default tag

thecodrr committed:
core: take into account boolean parameters in query

thecodrr committed:
web: fix infinite rerenders on checkout screen

thecodrr committed:
web: update lockfile

tarynlucia committed:
ui: correct grammar in vault creation message (issue #7497) (#8188)

Signed-off-by: Taryn Eng <taryn.eng@gmail.com>

thecodrr committed:
web: bump version to v3.2.0-beta.8

thecodrr committed:
core: fix search results sorting not working

thecodrr committed:
editor: fix search popup removed on link hover

ammarahm-ed committed:
mobile: make adding notebook a dialog

* make manage tags sheet a screen

ammarahm-ed committed:
mobile: release beta 4

thecodrr committed:
desktop: do not use native menu for tag suggestions on macos

thecodrr committed:
web: reduce trial2pro discount to 30%

thecodrr committed:
core: remove search results count

it was inaccurate and confused users

thecodrr committed:
web: bump version to v3.2.0-beta.9

01zulfi committed:
docs: add monograph size limit (#8206)

Signed-off-by: 01zulfi <85733202+01zulfi@users.noreply.github.com>

thecodrr committed:
core: fix group header for search results hidden

ammarahm-ed committed:
mobile: support android 16

ammarahm-ed committed:
mobile: fix pin and unpin from notifications

ammarahm-ed committed:
mobile: update deps

thecodrr committed:
global: regenerate package-lock files to fix CI (#8212)

01zulfi committed:
web: fix move notebook dialog (#8214)

* also use db.notebooks.parentId()

Signed-off-by: 01zulfi <85733202+01zulfi@users.noreply.github.com>

01zulfi committed:
core: fix notebooks parentId() tests (#8213)

Signed-off-by: 01zulfi <85733202+01zulfi@users.noreply.github.com>

01zulfi committed:
web: fix navigating to notebooks & tags from note item (#8211)

Signed-off-by: 01zulfi <85733202+01zulfi@users.noreply.github.com>

01zulfi committed:
editor: fix failing snapshot tests (#8215)

Signed-off-by: 01zulfi <85733202+01zulfi@users.noreply.github.com>

thecodrr committed:
web: fix ui glitch in note linking dialog

thecodrr committed:
web: fix tag navigation from editor

01zulfi committed:
web: fix archived notes showing in search results (#8216)

* web: fix archived notes showing in search results
Signed-off-by: 01zulfi <85733202+01zulfi@users.noreply.github.com>

* core: make notes param required in lookup.notesWithHighlighting
Signed-off-by: 01zulfi <85733202+01zulfi@users.noreply.github.com>

thecodrr committed:
core: fix small tokens ignored from query

thecodrr committed:
web: bump version to v3.2.0

ammarahm-ed committed:
mobile: release v3.2.0

thecodrr committed:
desktop: fix `'environment/DISABLE_WAYLAND' property does not match the
required schema`

thecodrr committed:
ci: disable snap on arm64

thecodrr committed:
desktop: change snap `base` to `core22`

thecodrr committed:
ci: publish snap beta

thecodrr committed:
web: bump version to v3.2.1-beta.0

thecodrr committed:
web: bump version to v3.2.1

thecodrr committed:
ci: release snap to stable track (instead of `release` track)

ammarahm-ed committed:
mobile: fix app crash on android 15 & 16

ammarahm-ed committed:
mobile: release v3.2.1

ammarahm-ed committed:
mobile: fix android toolbar hidden behind keyboard on android 15

thecodrr committed:
desktop: update better-sqlite3-multiple-ciphers to v11.10.0

thecodrr committed:
web: update sqlite to v3.49.2

thecodrr committed:
web: make trash actions multi selectable

thecodrr committed:
web: register fts5 extensions after db decryption

thecodrr committed:
web: bump version to v3.2.2

ammarahm-ed committed:
mobile: release v3.2.2

ammarahm-ed committed:
mobile: fix android toolbar

thecodrr committed:
core: update sqlite3-fts5-html to 0.0.4

thecodrr committed:
desktop: update sqlite3-fts5-html to 0.0.4

thecodrr committed:
web: update sqlite3-fts5-html to 0.0.4

ammarahm-ed committed:
mobile: convert sheets with input to screens or simple modals

01zulfi committed:
web: hide monograph list for not logged in users (#8258)

Signed-off-by: 01zulfi <85733202+01zulfi@users.noreply.github.com>

thecodrr committed:
web: only subscribe to note title if its generated

ammarahm-ed committed:
mobile: fix sqlite fts5

01zulfi committed:
editor: don't clear nodes in clear formatting option (#8260)

Signed-off-by: 01zulfi <85733202+01zulfi@users.noreply.github.com>

01zulfi committed:
web: fix scrolling toolbar also scrolling editor (#8256)

Signed-off-by: 01zulfi <85733202+01zulfi@users.noreply.github.com>

01zulfi committed:
web: fix update status height (#8267)

Signed-off-by: 01zulfi <85733202+01zulfi@users.noreply.github.com>

01zulfi committed:
core: don't generate headline title if title was set during note creation (#8266)

Signed-off-by: 01zulfi <85733202+01zulfi@users.noreply.github.com>

01zulfi committed:
web: fix note create option in quick open dialog (#8263)

* open new tab when query empty
* use editor-store methods for note creation

Signed-off-by: 01zulfi <85733202+01zulfi@users.noreply.github.com>
